### DISCUSSION TITLE
Never Free with twitter api？

### DISCUSSION THREAD

**Discussion by laohe-adam** [tweepy/tweepy#2159] [Questions]
### Question

tweepy.errors.Forbidden: 403 Forbidden
453 - You currently have access to a subset of Twitter API v2 endpoints and limited v1.1 endpoints (e.g. media post, oauth) only. If you need access to this endpoint, you may need a different access level. You can learn more here: https://developer.twitter.com/en/portal/product

### Checklist

- [ ] I have searched for duplicate questions.
- [ ] If applicable, I have removed any visible credentials from my code and/or screenshots.

**Comment by PhucHai123** [comment]
This is an answer on Stack Overflow for the same question as yours:

> This is from the official [Twitter developer website](https://developer.twitter.com/en).
> For write-only use cases and testing the Twitter API
> The free account only support posting 1500 tweets per month and does not allow reading tweets. To read tweets there are other paid plans which need to be purchased.

So basically, Twitter's API not free. You need to upgrade your subscription or apply for academic research to gain unlimited access (which I find impossible till this very moment, since the Twitter application form could be found nowhere).

If you need more details, here is the link to the question:
https://stackoverflow.com/questions/76297656/how-can-i-fix-the-403-forbidden-error-when-using-tweepy-to-access-the-twitter

**Comment by Harmon758** [comment]
I'm not sure what you're asking, and there's no information here besides an error message, without even the full traceback.
The fields that you've ignored in the form for questions aren't there for no reason.

**Comment by 0xgokuz** [comment]
@laohe-adam did you figure this out?

![Screenshot 2024-05-16 at 13 45 27](https://github.com/tweepy/tweepy/assets/155065525/4bae20ff-efe7-4973-9305-3347b7ce55bc)

I'm trying to post a tweet, and I'm getting the same error
